tests/utils/utils.go

```diff
@@ -444,15 +455,31 @@ func waitForTransaction(
 
 	if success {
 		if receipt.Status == types.ReceiptStatusFailed {
-			fmt.Println(TraceTransaction(ctx, subnetInfo, tx))
+			// Gomega will print the transaction trace
+			Expect(TraceTransaction(ctx, subnetInfo, receipt.TxHash)).Should(Equal(""))
 		}
-		Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
 	} else {
 		Expect(receipt.Status).Should(Equal(types.ReceiptStatusFailed))
 	}
 	return receipt
 }
 
+// Returns the first log in 'logs' that is successfully parsed by 'parser'
+// Errors and prints a trace of the transaction if no log is found.
+func GetEventFromLogsOrTrace[T any](
+	ctx context.Context,
+	receipt *types.Receipt,
+	subnetInfo interfaces.SubnetTestInfo,
+	parser func(log types.Log) (T, error),
+) T {
+	log, err := GetEventFromLogs(receipt.Logs, parser)
+	if err != nil {
+		// Gomega will print the transaction trace
+		Expect(TraceTransaction(ctx, subnetInfo, receipt.TxHash)).Should(Equal(""))
```

Comment:
i feel we can add this check inside the function `TraceTransaction`, and possibly indicate in the naming of the function that it will exit/fail

Reply:
Yeah, good suggestion